clusterctl/cmd/create_cluster.go

```diff
@@ -143,10 +143,28 @@ func getProvider(provider string) (clusterdeployer.ProviderDeployer, error) {
 	switch provider {
 	case "google":
 		return google.NewMachineActuator(google.MachineActuatorParams{})
-	case "terraform":
-		// TODO: Actually hook up terraform
-		return nil, errors.NotImplementedError
+	case "vsphere":
+		t, err := vsphere.NewMachineActuator("", nil, "")
+		if err != nil {
+			return nil, err
+		}
+		return &vsphereAdapter{t}, nil
 	default:
 		return nil, fmt.Errorf("Unrecognized provider %v", provider)
 	}
 }
+
+// Adapt the terraform methods calls since gcp/vsphere are not on the same page.
```

Comment:
should it be vsphere methods instead of terraform methods?

Reply:
Fixed.